app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/MapFragmentTest.kt

```diff
@@ -1,41 +1,52 @@
 package com.github.sdp_begreen.begreen.fragments
 
-import android.os.Bundle
+import android.Manifest
 import androidx.fragment.app.testing.launchFragmentInContainer
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.assertion.ViewAssertions.matches
-import androidx.test.espresso.matcher.ViewMatchers.withId
-import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.filters.LargeTest
+import androidx.test.rule.GrantPermissionRule
 import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.activities.MainActivity
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
+import kotlin.test.assertTrue
 
 @RunWith(AndroidJUnit4::class)
 @LargeTest
 class MapFragmentTest {
 
-    @Test
-    fun testMapFragment() {
-        launchFragmentInContainer<MapFragment>()
-        onView(withId(R.id.mapFragmentTextView)).check(matches(withText("Fragment where we can display the map")))
-    }
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
```

Comment:
I am not sure to understand why you are starting the `MainActivity` to test your `MapFragment` ?

I don't think that you need it.

Your tests will also likely pass once you've removed it.

Reply:
Well seen ! Thank you